crates/ruff/src/rules/perflint/rules/incorrect_dict_iterator.rs

```diff
@@ -0,0 +1,133 @@
+use ruff_text_size::TextRange;
+use rustpython_parser::ast::Expr;
+
+use crate::registry::AsRule;
+use ruff_diagnostics::{AlwaysAutofixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::prelude::Ranged;
+use rustpython_parser::ast;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks if a call to `dict.items()` uses only the keys or values
+///
+/// ## Why is this bad?
+/// Python dictionaries store keys and values in two separate tables. They can be individually
+/// iterated. Using .items() and discarding either the key or the value using _ is inefficient,
+/// when .keys() or .values() can be used instead:
+///
+/// ## Example
+/// ```python
+/// some_dict = {"a": 1, "b": 2}
+/// for _, val in some_dict.items():
+///     print(val)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// some_dict = {"a": 1, "b": 2}
+/// for val in some_dict.values():
+///     print(val)
+/// ```
+#[violation]
+pub struct IncorrectDictIterator {
+    subset: String,
+}
+
+impl AlwaysAutofixableViolation for IncorrectDictIterator {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let IncorrectDictIterator { subset } = self;
+        format!("When using only the {subset} of a dict use the `{subset}()` method")
+    }
+
+    fn autofix_title(&self) -> String {
+        let IncorrectDictIterator { subset } = self;
+        format!("Replace `.items()` with `.{subset}()`.")
+    }
+}
+
+/// PER8102
+pub(crate) fn incorrect_dict_iterator(checker: &mut Checker, target: &Expr, iter: &Expr) {
+    if let Expr::Call(ast::ExprCall { func, args, .. }) = iter {
+        let Expr::Attribute(ast::ExprAttribute { attr, value, ..  }) = func.as_ref() else {
+            return;
+        };
+        if attr != "items" {
+            return;
+        }
+        if !args.is_empty() {
+            return;
+        }
+        if let Expr::Tuple(ast::ExprTuple {
+            elts,
+            range: tup_range,
+            ..
+        }) = target
+        {
+            if elts.len() != 2 {
```

Comment:
You should be able to remove the length check here because your match pattern only matches for exactly two elements